File: pubsub/src/main/java/glitch/pubsub/object/json/CommerceMessage.java
package glitch.pubsub.object.json;

import com.google.gson.annotations.SerializedName;
import lombok.Data;

import java.time.Instant;

@Data
public class CommerceMessage {
    @SerializedName("user_name")
    private final String username;
    private final String displayName;
    private final String channelName;
    private final Long userId;
    private final Long channelId;
    private final Instant time;
    private final String itemImageUrl;
    private final String itemDescription;
    private final boolean supportsChannel;
    private final String purchaseMessage;
}

File: pubsub/src/main/kotlin/glitch/pubsub/object/json/CommerceMessage.kt
package glitch.pubsub.`object`.json

import com.google.gson.annotations.SerializedName
import java.time.Instant

data class CommerceMessage(
        @SerializedName("user_name")
        val username: String, val displayName: String,
        val channelName: String, val userId: Long,
        val channelId: Long, val time: Instant,
        val itemImageUrl: String, val itemDescription: String,
        val isSupportsChannel: Boolean, val purchaseMessage: String
)
